### PR TITLE
fix bug 7068 that can't render two slots at the same time 

### PR DESCRIPTION
**Pull request by JerryWu1234**
## Changes

Loading two slots in the async pattern will lead to a component of nested which can not be loaded.  
fix bug #7068

### CODE REVIEW

File: packages/astro/src/runtime/server/render/astro/instance.ts [hunk 1]
```diff
@@ -31,8 +31,9 @@ export class AstroComponentInstance {
 		this.factory = factory;
 		this.slotValues = {};
 		for (const name in slots) {
-			const value = slots[name](result);
-			this.slotValues[name] = () => value;
+			// collections-renderentry.test.js this units needs to early get the css from the children component and the slot  
+			slots[name](result)
+			this.slotValues[name] = () => slots[name](result);
```
Comment (matthewp) [inline packages/astro/src/runtime/server/render/astro/instance.ts:36]: Calling this twice feels wrong to me. I can believe it fixes the test but it's puzzling why that would be.

Comment (wulinsheng123) [reply packages/astro/src/runtime/server/render/astro/instance.ts:36]: I have tried a lot but I still haven’t found the way it broke that unit test.
It's very weird that it only breaks that one-unit test

Comment (wulinsheng123) [reply packages/astro/src/runtime/server/render/astro/instance.ts:35]: https://github.com/withastro/astro/blob/31cbf4357e866abab62bb4ccf949f5a59aec06ac/packages/astro/test/units/dev/collections-renderentry.test.js#L157
There isn’t CSS, but the CSS style tag even appears in the bundle.

Comment (JerryWu1234) [reply packages/astro/src/runtime/server/render/astro/instance.ts:35]: @matthewp coule help me check this

**Comment by JerryWu1234** [comment]
it's weird 
everything is going well.
however, I added some tests for this part.

**Comment by matthewp** [comment]
Hey @JerryWu1234, thanks as always. Going to close since the PR is old. Happy to have a new one if you can figure out the test problem, thanks!

**Pull request closed without merging**